Documentation/Signoffs/PLC.md

```diff
@@ -0,0 +1,67 @@
+
+## Constraints 
+
+**The PLC:** 
+
+* **Must have at least 6 Digital Input pins**
+
+^ These are for the 3 door locks and sensors in order to be read by the PLC.
+
+* **Must have at least 6 Digital Output pins**
+
+^ These are also for the 3 door lock and sensors in order to be commanded by the PLC.
+
+* **Must have at least 3 24VDC Outputs**
+
+^ These are going to be used for the motor driver.
+
+From this, the input module must have at least 6 ports, and the output module must have at least 9 ports 
+
+* **Must be able to fit in a 1 square-foot area**
+
+^ This is for clearance in the area where we are going to house all of our controllers and electronics, aside the motor. 
+
+* **Will be in accordance with IEC-61131-2(6.4.4.3) which states:**
+
+"When the short circuit (for a voltage output) or the open circuit (for a current output) is realized, no physical damage or abnormal phenomenon shall be detected." 
+
+This will be accomplished by having the GFCI in between our hardware and the power supply. 
+
+^ Source: http://rpa.energy.mn/wp-content/uploads/2017/03/IEC-61131-2-Programmable-controllers-Equipment-Requirements-and-Tests.pdf 
+
+* **Must be able to communicate with our PC**
+^ (The information about communication are also discussed in the Sensor System signoff)
+To remove data from the computer for use, a program will be added to the PC so when the drive is plugged in, the information is downloaded automatically. The port will be behind a locked compartment so students cannot access the data.
+
+To connect the Computer to the PLC, a USB NIC will be needed to provide an extra Ethernet port. The PLC uses Profinet, which is a communication protocol used to collect information and control a PLC through Profibus. The Computer will use an application called TIA Portal which allows connection and communication with the PLC and functions as a controller for S7 1200 and 1500 series PLCs. To use the TIA portal software, the PLC needs a minimum of 16 GB of RAM and an Intel Core i5-8400H or stronger CPU. The ThinkCentre provides enough RAM and a strong enough CPU to handle the program. The program will be created in the TIA portal and loaded to the memory on the PLC. TIA portal also provides diagnostics of the PLC through the software. The PLC the team is using is called a SIMATIC S7 1200 from Siemens and requires certain types of memory card. The three options of memory for the PLC differ only by capacity. The three options for memory size are 2, 4, and 24 MB. The team is using a 4MB memory card for the PLC since the 2MB is out of production. Issues have arrived with a USB NIC not being recognized by TIA Portal. If the software does not recognize the NIC, the way to fix the issue is to enter the IP address manually. The TIA Portal SIMATIC Step 7 program allows the use of an OPC UA Server.
```

Comment:
This will not allow you to communicate with the PC. As I have said. TIA portal doesn't facilitate communication.